src/components/chatItems.tsx

```diff
@@ -239,6 +229,26 @@ export function UserChatItem(props: {
   chatItem: StreamingStepInput;
   AIname?: string;
 }) {
+  let graphedData: GraphData | null = null;
+  let content = props.chatItem.content;
+
+  if (props.chatItem.role === "function") {
+    graphedData = extractGraphData(props.chatItem.content);
+    try {
+      const functionJsonResponse = JSON.parse(props.chatItem.content) as Json;
+      if (functionJsonResponse && typeof functionJsonResponse === "object") {
+        content = convertToRenderable(
+          functionJsonResponse,
+          `${functionNameToDisplay(props.chatItem?.name ?? "")} result`,
+        );
+      }
+    } catch {
+      // If not JSON, then there may be a summary
+      content = props.chatItem.summary ?? props.chatItem.content;
+    }
+  }
+  const [tabOpen, setTabOpen] = useState<"table" | "graph">("table");
```

Comment:
This logic is a bit confusing for me, you switch to the summary if the attempt to JSON the content errors?

Reply:
Yep, we only summarise non-JSON responses, so if it fails to convert to JSON, there may be a summary.

If it's JSON, there's definitely no summary, so no point in checking